### PR TITLE
Initial Countly work

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/15608

### CODE REVIEW

File: src/CountlyAnalytics.ts [hunk 1]
```diff
@@ -0,0 +1,941 @@
...
+        });
+
+        // if this event can be sent anonymously and we are disabled then dispatch it right away
+        if (this.disabled && anonymous) {
+            await this.request({ device_id: randomString(64) });
+        }
+    }
+}
+
+window.mxCountlyAnalytics = CountlyAnalytics;
```
Comment (dbkr) [inline src/CountlyAnalytics.ts:941]: doesn't look like this is used anywhere?

**Review by dbkr — COMMENTED** [review]
otherwise looks plausible!

File: src/CountlyAnalytics.ts (hunk 1 above)
Comment (t3chguy) [reply src/CountlyAnalytics.ts:941]: We expose it on Window for debugging, like we do with most other things, to make it easily reachable from a console

File: src/CountlyAnalytics.ts [hunk 2]
```diff
@@ -0,0 +1,941 @@
...
+
+// polyfill textencoder if necessary
+import * as TextEncodingUtf8 from 'text-encoding-utf-8';
+let TextEncoder = window.TextEncoder;
+if (!TextEncoder) {
+    TextEncoder = TextEncodingUtf8.TextEncoder;
+}
+
+const INACTIVITY_TIME = 20; // seconds
+const HEARTBEAT_INTERVAL = 5_000;
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/CountlyAnalytics.ts:34]: ms yes, readability, cleaner as 5s imo

**Review by nadonomy — APPROVED** [review]

**Review by dbkr — APPROVED** [review]
ag - gotcha, thanks

**Pull request merged**